### PR TITLE
Features/chocolatey source added state

### LINKED ISSUE #58588: [FEATURE REQUEST] Add a way to use Chocolatey repositories with authentication
**Is your feature request related to a problem? Please describe.**
We want to use a custom Chocolatey repository that requires authentication with Salt states. Currently, there is no way to use authentication with states. With the Chocolatey module, it is possible to add a repository with authentication thanks to `chocolatey.add_source`.

**Describe the solution you'd like**
Add an `source_added` function to the chocolatey state that will add a Chocolatey repository with authentication if needed.

**Describe alternatives you've considered**
Add `username` and `password` as arguments to all the functions that use the `source` argument.

**Additional context**
A PR has been submitted to address this issue: #58590

### PR DESCRIPTION
**Pull request by TGuimbert**
### What does this PR do?
Add the `source_added` function for the Chocolatey state that match the `add_source` abilities of the Chocolatey module. This add a way to use custom Chocolatey repository with authentication. To match the behavior of the `installed` function, I also added the function `list_sources` in the module.

### What issues does this PR fix or reference?
Fixes: #58588 

### Merge requirements satisfied?
- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

Is the doc from the functions enough? 

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by TGuimbert** [comment]
Should I remove legacy Python syntax from the chocolatey module and state as asked by `ci/pre-commit`?

**Review by ArthurREGNARD — DISMISSED** [review]

File: salt/states/chocolatey.py [hunk 1]
```diff
@@ -460,3 +452,81 @@ def upgraded(
     ret["changes"] = salt.utils.data.compare_dicts(pre_install, post_install)
 
     return ret
+
+
+def source_added(name, source_location, username=None, password=None, force=False):
```
Comment (twangboy) [inline salt/states/chocolatey.py:457]: To follow the Salt naming convention, please change this to `source_present`

**Review by twangboy — CHANGES_REQUESTED** [review]
This looks great. Thanks for the PR. One requested change.

File: salt/states/chocolatey.py (hunk 1 above)
Comment (TGuimbert) [reply salt/states/chocolatey.py]: Thank you for your feedback. It should be better now.

**Review by twangboy — APPROVED** [review]

**Comment by Slemaire-PNI** [comment]
Observation: Seems like it lacks the option to set priority for each source, which is pretty important.  Also in the patch notes for 3002, this state function was reported to be source_added but it's actually source_present.

**Pull request merged**